sentry/src/main/java/io/sentry/util/UrlUtils.java

```diff
@@ -0,0 +1,186 @@
+package io.sentry.util;
+
+import io.sentry.ISpan;
+import io.sentry.protocol.Request;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class UrlUtils {
+
+  private static final @NotNull Pattern AUTH_REGEX = Pattern.compile("(.+://)(.*@)(.*)");
+
+  public static @Nullable UrlDetails convertUrlNullable(final @Nullable String url) {
+    if (url == null) {
+      return null;
+    }
+
+    return convertUrl(url);
+  }
+
+  public static @NotNull UrlDetails convertUrl(final @NotNull String url) {
```

Comment:
`l`: I'm wondering if `parse()` or `parseUrl()` would suit better here

Reply:
No preference here. Can rename it if you want.

Reply:
Yeah I think it would fit better, but I'm definitely nit-picking here - if it's too much effort to refactor leave it as is 😅

Reply:
Updated